### PR TITLE
Implement AI upscaling in Dolphin

### PR DESCRIPTION
**Pull request by delroth**
Inspired by bsnes' AI upscaling described in https://www.theverge.com/2019/4/18/18311287/ai-upscaling-algorithms-video-games-mods-modding-esrgan-gigapixel

### CODE REVIEW

**Comment by RinMaru** [comment]
ill give it a test hope it doesnt need RTX

**Comment by ExtremeDude2** [comment]
> How does it work and how do you enable it?

It's clearly labeled inthe graphics settings ;)

**Comment by RinMaru** [comment]
> > How does it work and how do you enable it?
> 
> It's clearly labeled inthe graphics settings ;)

yea i noticed but dont see anything happening

**Comment by Miksel12** [comment]
Wow! The difference is insane ;)

**Comment by jordan-woyak** [comment]
@delroth How many hours did you spend tweaking the algorithm?

**Comment by mimimi085181** [comment]
How big is the training data and how can people contribute?

**Comment by JMC47** [comment]
Had to try this myself.  I can't believe the results.

Before

AI2

After

AI1

Not only is it clearer, the AI even translated the game!

**Comment by jordan-woyak** [comment]
@delroth This is causing issues with bloom effects in many games (e.g. SMG1/2). That definitely needs to be resolved before merge.

**Comment by JMC47** [comment]
Example of what's going on in Super Mario Galaxy 1/2

image

**Comment by RinMaru** [comment]
i dont get it isnt that just higher internal res?

> Had to try this myself. I can't believe the results.
> 
> Before
> 
> AI2
> 
> After
> 
> AI1
> 
> Not only is it clearer, the AI even translated the game!

**Comment by Miksel12** [comment]
> i dont get it isnt that just higher internal res?

What? No, this is waaaay more advanced ;)

**Comment by Craftyawesome** [comment]
> i dont get it isnt that just higher internal res?

The internal resolution feature cannot translate games.

Is there an option to turn the translation off? Speedrunners will want to use the faster text many Japanese games use.

**Comment by RinMaru** [comment]
I think I get it now it hasnt been implemented yet this is just the config option lables *feels stupid*

File: Source/Core/DolphinQt/Config/Graphics/EnhancementsWidget.cpp [hunk 1]
```diff
@@ -282,9 +282,9 @@ void EnhancementsWidget::SaveSettings()
 void EnhancementsWidget::AddDescriptions()
 {
   static const char TR_INTERNAL_RESOLUTION_DESCRIPTION[] =
-      QT_TR_NOOP("Controls the rendering resolution. A high resolution greatly improves visual "
+      QT_TR_NOOP("Controls the AI upscaling goodness. A high resolution greatly improves visual "
```
Comment (spycrab) [inline Source/Core/DolphinQt/Config/Graphics/EnhancementsWidget.cpp]: Does not adequately describe the advanced algorithms used to achieve these marvellous results.

Comment (Techjar) [reply Source/Core/DolphinQt/Config/Graphics/EnhancementsWidget.cpp]: Yeah we'll have to add a scroll box so we can fit the entire whitepaper in.

Comment (spycrab) [reply Source/Core/DolphinQt/Config/Graphics/EnhancementsWidget.cpp]: We should look into adding a PDF renderer into Externals: https://wiki.qt.io/Handling_PDF#Rendering

**Comment by MayImilae** [comment]
Please don't merge this.

**Comment by delroth** [comment]
With so much resistance to this change on the main Dolphin repo I think I'm just going to contribute this to Ishiiruka instead. Too bad.

**Pull request closed without merging**